### PR TITLE
fix(editor): Update SSH repository URL validation regex to support dots in usernames

### PR DESCRIPTION
**Pull request by AbdulTawabJuly**
### Summary

Updates the SSH repository URL validation regex in the Source Control settings to properly handle dots (.) in usernames. The previous regex pattern `[\w-]+@` was too restrictive and didn't allow dots in the username portion of SSH URLs, which are valid in many Git hosting services.

**Changes:**
- Modified regex pattern from `[\w-]+@` to `[\w.-]+@` in `SettingsSourceControl.vue`
- This allows dots in usernames while maintaining security and validation

**Testing:**
- Added comprehensive unit tests for the validation rules
- Tested with various SSH URL formats including dots in usernames
- All tests pass successfully

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included.
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/frontend/editor-ui/src/features/sourceControl.ee/views/SettingsSourceControl.vue [hunk 1]
```diff
@@ -174,7 +174,7 @@ const repoUrlValidationRules = computed<Array<Rule | RuleGroup>>(() => {
 			name: 'MATCH_REGEX',
 			config: {
 				regex:
-					/^(?:git@|ssh:\/\/git@|[\w-]+@)(?:[\w.-]+|\[[0-9a-fA-F:]+])(?::\d+)?[:\/][\w\-~.]+(?:\/[\w\-~.]+)*(?:\.git)?(?:\/.*)?$/,
+					/^(?:git@|ssh:\/\/git@|[\w.-]+@)(?:[\w.-]+|\[[0-9a-fA-F:]+])(?::\d+)?[:\/][\w\-~.]+(?:\/[\w\-~.]+)*(?:\.git)?(?:\/.*)?$/,
```
Comment (konstantintieber) [inline packages/frontend/editor-ui/src/features/sourceControl.ee/views/SettingsSourceControl.vue:177]: The change looks good, can you add a test case for the `should test repo URLs` in the `for ssh connection` using a username containing a dot, so we can ensure that we keep supporting this?

Comment (AbdulTawabJuly) [reply packages/frontend/editor-ui/src/features/sourceControl.ee/views/SettingsSourceControl.vue:177]: Thanks for the feedback! I've added comprehensive test cases for SSH URLs with dots in usernames and also updated the regex pattern to handle the edge cases properly.

`What I've done:
`

- Added 7 new test cases covering various SSH URL formats with dots in usernames:
   - git.user@github.com:user/repository.git
   - user.name@github.com:user/repository.git
   - ssh://git.user@mydomain.example:2224/gitolite-admin
   - And 4 more variations

- Updated the regex pattern

- Edge cases now handled:
   - SSH protocol with dots: ssh://git.user@domain.com:port/repo.git (was failing before)
   - Standard format with dots: git.user@domain.com:repo.git (was failing before)
   - Various Git hosting services: GitHub, GitLab, Azure DevOps, custom servers
   - Different port configurations: URLs with and without custom ports

Comment (konstantintieber) [reply packages/frontend/editor-ui/src/features/sourceControl.ee/views/SettingsSourceControl.vue:177]: I'm sorry this is more than I would've expected and now I'm confused.
Why did you have to add another change to the regex, when the previous change added in https://github.com/n8n-io/n8n/pull/20666/commits/385f131627743e55e5b5ecd10989b77c3de151ca already added support for usernames with dots?

Comment (AbdulTawabJuly) [reply packages/frontend/editor-ui/src/features/sourceControl.ee/views/SettingsSourceControl.vue:177]: I apologize for the confusion! You're absolutely right. I thought the SSH protocol format with dots in usernames might be an edge case, but it was unnecessary to modify the regex further.

I've reverted the regex back to the previous version and added 6 test cases to ensure the existing functionality works correctly

Comment (konstantintieber) [reply packages/frontend/editor-ui/src/features/sourceControl.ee/views/SettingsSourceControl.vue:177]: Since one test containing a username with a `.` is enough to ensure that this functionality keeps being supported, what do you think about only adding one test case?

**Review by konstantintieber — APPROVED** [review]
Thx for the diligent work 💯
